src/gsi/crtm_interface.f90

```diff
@@ -2046,11 +2122,21 @@ subroutine call_crtm(obstype,obstime,data_s,nchanl,nreal,ich, &
               do ii=1,n_clouds_fwd_wk
                 !cloud_cont(k,ii)=cloud(kk2,ii)*kgkg_kgm2 
                  cloud_cont(k,ii)=cloud(kk2,ii)*c6(k)
-                 if (imp_physics==11 .and. lprecip_wk .and.  cloud_cont(k,ii) > 1.0e-6_r_kind) then
+                !Option for Thompson added
+                 if (imp_physics==11 .or. imp_physics==08 .and. lprecip_wk .and.  cloud_cont(k,ii) > 1.0e-6_r_kind) then
                     cloud_efr(k,ii)=cloudefr(kk2,ii)
                  else
                     cloud_efr(k,ii)=zero
                  endif
+                 if (trim(cloud_names_fwd(ii))=='ni' .and.  atmosphere(1)%temperature(k)<t0c) then
+                     cloud_cont(k,ii)=max(1.001_r_kind*1.0E-6_r_kind, cloud_cont(k,ii))
+                     cloud_efr(k,ii)=max(5.001_r_kind, cloud_efr(k,ii))
+                 endif
+                 if (trim(cloud_names_fwd(ii))=='nr' .and.  atmosphere(1)%temperature(k)<t0c) then
+                     cloud_cont(k,ii)=max(1.001_r_kind*1.0E-6_r_kind, cloud_cont(k,ii))
+                     cloud_efr(k,ii)=max(5.001_r_kind, cloud_efr(k,ii))
+                 endif
+
```

Comment:
These lines are not needed and should be removed.
We do not need to check number concentration since CRTM does not take number concentration as input.  We just need number concentration (ni and nr) for effective radius calculation